examples/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-if(WITH_OTLP)
+if(WITH_OTLP_GRPC OR WITH_OTLP_HTTP)
```

Comment:
Do we really need that level of granularity, can one flag build both?

Reply:
I think it is rare for one build to have both, so 2 flags make sense. Or we could introduce a third flag `WITH_OTLP` to include both but then it seems over complicated.

Reply:
Sometimes we want only one of `WITH_OTLP_GRPC` and `WITH_OTLP_HTTP` .
Set `WITH_OTLP=ON` will try to turn `WITH_OTLP_GRPC` on when grpc is found and  turn `WITH_OTLP_HTTP` on when both libcurl and nlohmann_json are found in this PR.
The `WITH_OTLP` option is just for backward compatibility, and the actually options take effect is  `WITH_OTLP_GRPC` and `WITH_OTLP_HTTP` now.